src/connectors/menu/connectMenu.js

```diff
@@ -137,74 +137,39 @@ export default function connectMenu(renderFn, unmountFn = noop) {
         return this.isShowingMore ? showMoreLimit : limit;
       },
 
-      refine(helper) {
-        return facetValue => {
+      init(initOptions) {
+        const { helper, createURL, instantSearchInstance } = initOptions;
+
+        this._createURL = facetValue =>
+          createURL(helper.state.toggleRefinement(attribute, facetValue));
+
+        this._refine = function(facetValue) {
           const [refinedItem] = helper.getHierarchicalFacetBreadcrumb(
             attribute
           );
           helper
             .toggleRefinement(attribute, facetValue ? facetValue : refinedItem)
             .search();
         };
-      },
-
-      init({ helper, createURL, instantSearchInstance }) {
-        this.cachedToggleShowMore = this.cachedToggleShowMore.bind(this);
-
-        this._createURL = facetValue =>
-          createURL(helper.state.toggleRefinement(attribute, facetValue));
-
-        this._refine = this.refine(helper);
 
         renderFn(
           {
-            items: [],
-            createURL: this._createURL,
-            refine: this._refine,
+            ...this.getWidgetRenderState(initOptions),
             instantSearchInstance,
-            canRefine: false,
-            widgetParams,
-            isShowingMore: this.isShowingMore,
-            toggleShowMore: this.cachedToggleShowMore,
-            canToggleShowMore: false,
           },
           true
         );
       },
 
-      render({ results, instantSearchInstance }) {
-        const facetValues = results.getFacetValues(attribute, { sortBy });
-        const facetItems =
-          facetValues && facetValues.data ? facetValues.data : [];
-
-        const items = transformItems(
-          facetItems
-            .slice(0, this.getLimit())
-            .map(({ name: label, path: value, ...item }) => ({
-              ...item,
-              label,
-              value,
-            }))
-        );
+      render(renderOptions) {
+        const { instantSearchInstance } = renderOptions;
 
-        this.toggleShowMore = this.createToggleShowMore({
-          results,
-          instantSearchInstance,
-        });
+        this.toggleShowMore = this.createToggleShowMore(renderOptions);
```

Comment:
IMO, it's better to explicitly pass `results` and `instantSearchInstance`.

Reply:
I did it this way for readability. The other widgets (`refinementList` and `hierarchicalMenu`) are doing it this way too but as they need the whole `renderOptions` and not just these two variables, I can revert it back if you think it's better.